google/cloud/datastore/query.py

```diff
@@ -35,6 +41,100 @@
 )
 
 
+class BaseFilter(ABC):
+    """Base class for Filters"""
+
+    @abc.abstractmethod
+    def build_pb(self, container_pb=None):
+        """Build the protobuf representation based on values in the Filter."""
+
+
+class PropertyFilter(BaseFilter):
+    """Class representation of a Property Filter"""
+
+    def __init__(self, property_name, operator, value):
+        if property_name == "__key__" and not isinstance(value, Key):
+            raise ValueError('Invalid key: "%s"' % value)
+        if Query.OPERATORS.get(operator) is None:
+            error_message = 'Invalid expression: "%s"' % (operator,)
+            choices_message = "Please use one of: =, <, <=, >, >=, !=, IN, NOT_IN."
+            raise ValueError(error_message, choices_message)
+        self.property_name = property_name
+        self.operator = operator
+        self.value = value
+
+    def build_pb(self, container_pb=None):
+        """Build the protobuf representation based on values in the Property Filter."""
+        container_pb.op = Query.OPERATORS.get(self.operator)
+        container_pb.property.name = self.property_name
+        if self.property_name == "__key__":
+            key_pb = self.value.to_protobuf()
+            container_pb.value.key_value.CopyFrom(key_pb._pb)
+        else:
+            helpers._set_protobuf_value(container_pb.value, self.value)
+        return container_pb
+
+    def __repr__(self):
+        return f"<{self.property_name} {self.operator} '{self.value}'>"
+
+
+class BaseCompositeFilter(BaseFilter):
+    """Base class for a Composite Filter. (either OR or AND)."""
+
+    def __init__(
+        self,
+        operation=query_pb2.CompositeFilter.Operator.OPERATOR_UNSPECIFIED,
+        filters=None,
+    ):
+        self.operation = operation
+        if filters is None:
+            self.filters = []
+        else:
+            self.filters = filters
+
+    def __repr__(self):
+        repr = f"op: {self.operation}\nFilters:"
+        for filter in self.filters:
+            repr += f"\n\t{filter}"
+        return repr
+
+    def build_pb(self, container_pb=None):
```

Comment:
For reference, the node.js version of this is here:

https://github.com/googleapis/nodejs-datastore/pull/1061/files#diff-87404c91a22dbabcc334f9a5288341281cb2b6625533e1207efcac16dc1e2ffdR147